pkgs/io_file/lib/src/vm_posix_file_system.dart

```diff
@@ -2,12 +2,38 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:ffi';
 import 'dart:io' as io;
+import 'dart:math';
+import 'dart:typed_data';
 
+import 'package:ffi/ffi.dart' as ffi;
+import 'package:meta/meta.dart';
 import 'package:stdlibc/stdlibc.dart' as stdlibc;
 
 import 'file_system.dart';
 
+// The maximum number of bytes to read in a single call to `read`.
+//
+// On macOS, it is an error to call
+// `read/_read(fildes, buf, nbyte)` with `nbyte >= INT_MAX`.
+//
+// The POSIX specification states that the behavior of `read` is
+// implementation-defined if `nbyte > SSIZE_MAX`. On Linux, the `read` will
+// transfer at most 0x7ffff000 bytes and return the number of bytes actually.
+// transfered.
+//
+// A smaller value has the advantage of making vm-service clients
+// (e.g. debugger) more responsive.
+//
+// A bigger value reduces the number of system calls.
+@visibleForTesting
```

Comment:
You can hide them from the package by not exporting them in `lib/posix_file_system.dart`: `export 'src/vm_posix_file_system.dart' show PosixFileSystem;`. (Apparently this can't be combined with `if`, so another option is to move these to a different file.)

I don't feel super strong about it, but we can avoid this annotation.

Reply:
Let me see how this goes. When I write the Windows implementation then I might more this around anyway.